docs/extension.md

```diff
@@ -0,0 +1,215 @@
+      - X-Custom-Header1
+      - X-Custom-Header2
+    max_age: 86200
+```
+
+## Properties Overview
+`x-kusk` extension can be applied at (not exclusively):
+1. Top level of an OpenAPI spec:
+```yaml
+openapi: 3.0.2
+info:
+  title: Swagger Petstore - OpenAPI 3.0
+x-kusk:
+  hosts: 
+  - "example.org"
+  disabled: false
+  cors:
+    ...
+
+```
+
+2. Path level:
+```yaml
+openapi: 3.0.2
+info:
+  title: Swagger Petstore - OpenAPI 3.0
+paths:
+  /pet:
+    x-kusk:
+      disabled: true # disables all /pet endpoints
+    post:
+      ...
+```
+3. Method (operation) level:
+```yaml
+  openapi: 3.0.2
+  info:
+    title: Swagger Petstore - OpenAPI 3.0
+  paths:
+    /pet:
+      post:
+        x-kusk:
+          upstream: # routes the POST /pet endpoint to a Kubernetes service
+            service:
+              namespace: default
+              service: petstore
+              port: 8000
+        ...
+```
+
+## Property Overriding/inheritance
+
+  `x-kusk` extension at the operation level takes precedence, i.e. overrides, what's specified at the path level, including the `disabled` option.
+  Likewise, the path level settings override what's specified at the global level.
+
+  If settings aren't specified at a path or operation level, it will inherit from the layer above. (Operation > Path > Global)
+
+## Available properties
+
+### disabled
+
+This boolean property allows you to disable the corresponding path/operation, allowing you to "hide" internal operations
+from being published to end users.
+
+When set to true at the top level all paths will be hidden; you will have to override specific paths/operations with
+`disabled: false` to make those operations visible.
+
+### hosts
+
+This string array property configures hosts list the Gateway will listen traffic for.
+
+### cors
+
+The `cors` object sets properties for configuring [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) for your API.
+
+| Name | Description |
+| :---: | :--- |
+| `origins` | list of HTTP origins accepted by the configured operations
+| `methods` | list of HTTP methods accepted by the configured operations
+| `headers` | list of HTTP headers accepted by the configured operations
+| `expose_headers` | list of HTTP headers exposed by the configured operations
+| `credentials` | boolean flag for requiring credentials
+| `max_age` | indicates how long results of a preflight request can be cached
+
+
+### qos
+
+Options for configuring QoS settings, such as retries and timeouts.
+
+| Name | Description |
+| :---: | :--- |
+| `retries` | maximum number of retries (0 by default)
+| `request_timeout` | total request timeout (in seconds)
+| `idle_timeout` | timeout for idle connections (in seconds)
+
+Please see the documentation for each individual generator to see which of these properties they support and how they apply.
+
+### upstream
+
+This setting configures where the traffic goes. `service` and `host` are available and are mutually exclusive.
```

Comment:
Extending, Service is K8s service that exists in the local cluster k8s,  while host - any hostname possible, even if outside of the cluster.